src/tls.c

```diff
@@ -186,6 +190,27 @@ enftun_tls_connect(struct enftun_tls* tls,
     return rc;
 }
 
+#ifdef USE_PSOCK
+int
+enftun_tls_proxy_connect(struct enftun_tls* tls,
+                         const char** hosts,
+                         const char* port)
+{
+    int rc;
+
+    rc = enftun_tcp_proxy_connect_any(&tls->sock, hosts, port);
+    if (rc < 0)
+        goto out;
+
+    rc = enftun_tls_handshake(tls);
+    if (rc < 0)
+        enftun_tcp_close(&tls->sock);
```

Comment:
Should be `enftun_tcp_proxy_connect_close`.  The same change is needed in `enftun_tls_disconnect`, if using `tcp_proxy`.

This would require tracking whether we connected with tcp or tcp_proxy. Let's discuss reworking the TCP interface to make this easier.

Reply:
Could we create a `enftun_connection_open` that does either `enftun_tcp_connect` or `enftun_psock_connect` based on USE_PSOCK and then same for a `enftun_connection_close`?

It seems that then the only ifdefs (other than associate with conn_state) we would have is inside the enftun_connection_{open/close} functions.